src/utils/react-query-factory.js

```diff
@@ -0,0 +1,93 @@
+// eslint-disable-next-line import/no-extraneous-dependencies
+import { createQueryKeyStore } from '@lukemorales/query-key-factory';
+import { fetchCourseMetadata } from '../components/app/routes/queries/courseMetadata';
+import { fetchUserEntitlements } from '../components/app/routes/queries/userEntitlements';
+import { fetchCanRedeem } from '../components/app/routes/queries/canRedeemCourse';
+import { fetchEnterpriseLearnerData } from '../components/app/routes/queries/enterpriseLearner';
+import { fetchEnterpriseCourseEnrollments } from '../components/app/routes/queries/enterpriseCourseEnrollments';
+import { fetchBrowseAndRequestConfiguration } from '../components/app/routes/queries/subsidies/browseAndRequest';
+import { fetchRedeemablePolicies } from '../components/app/routes/queries/subsidies/policies';
+import { fetchSubscriptions } from '../components/app/routes/queries/subsidies/subscriptions';
+import { fetchCouponCodes } from '../components/app/routes/queries/subsidies/couponCodes';
+import { fetchEnterpriseOffers } from '../components/app/routes/queries/subsidies/enterpriseOffers';
+import { fetchEnterpriseCuration } from '../components/app/routes/queries/contentHighlights';
+
+// The keys that map to enterpriseQueryKeys that is used to define specific queries
+// and used to specifically invalidate queries
+const enterprise = createQueryKeyStore('enterprise', {
+  enterpriseCustomer: (enterpriseUuid) => ({
+    queryKey: [enterpriseUuid],
+    contextQueries: {
+      course: (courseKey) => ({
+        queryKey: [courseKey],
+        contextQueries: {
+          contentMetadata: {
+            queryKey: null,
+            queryFn: async ({ queryKey }) => fetchCourseMetadata(queryKey[2], queryKey[4]),
+          },
+          canRedeem: (availableCourseRunKeys) => ({
+            queryKey: [availableCourseRunKeys],
+            queryFn: async ({ queryKey }) => fetchCanRedeem(queryKey[2], availableCourseRunKeys),
+          }),
+        },
+      }),
+      user: {
+        queryKey: ['entitlements'],
+        queryFn: async () => fetchUserEntitlements(),
+      },
```

Comment:
[suggestion] Might make sense to move this `user` query to no longer be nested under `enterpriseCustomer` since it does not interact at all with a specific enterprise customer. Instead, it can be a standalone top-level key.

Related, no strong feelings here, but it might be a bit preferable to define enterprise vs. user as 2 separate query key factories that are merged together with `createQueryKeys` and `mergeQueryKeys`.

```jsx
const enterprise = createQueryKeys('enterprise', { ... });
const user = createQueryKeys('user', {
  entitlements: null,
});
const queries = mergeQueryKeys(enterprise, user);
```

The benefit of splitting them out is that they can be defined a bit more granularly and keeps non-enterprise related queries outside of the `enterprise` query key factory.

Don't feel super strongly here; curious to hear what you think!

Reply:
This is a great idea, there are a couple of attributes outside of `enterpriseCustomer` that can be split off and merged 👍🏽

Reply:
@brobro10000 Just a heads up, if we go with this more granular route, it's using `createQueryKeys`, not `createQueryKeyStore` 👀 

I believe `createQueryKeyStore` is meant for when you are defining a single query object?